wandb/sdk/artifacts/artifact.py

```diff
@@ -638,6 +651,45 @@ def updated_at(self) -> str:
         assert self._created_at is not None
         return self._updated_at or self._created_at
 
+    @property
+    def ttl(self) -> Optional[datetime.timedelta]:
+        """Artifact TTL(time to live).
+
+        Once the artifact's created_at is older than the ttl, the artifact is marked as deleted.
+        To turn off previously set artifact TTL, set TTL to wandb.ArtifactTTL.DELETE
+        EX: Artifact.created_at = 1/1/2000, ttl=10 days.
+            This artifact will be marked for deletion on 1/11/2000.
+        """
+        if (
+            self._ttl_duration_seconds is None
+            or self._ttl_duration_seconds == ArtifactTTL.DELETE.value
+        ):
+            return None
+        return datetime.timedelta(seconds=self._ttl_duration_seconds)
+
+    @ttl.setter
+    def ttl(self, ttl: Union[datetime.timedelta, ArtifactTTL]) -> None:
+        """Artifact TTL(time to live).
+
+        Note:
+            This ttl is set on the artifact version level.
+
+        Arguments:
+            ttl: How long the artifact will remain active from its creation.
+            To turn off previously set artifact TTL, set TTL to wandb.ArtifactTTL.DELETE
+        """
+        if self.type == "wandb-history":
+            raise ValueError("Cannot set artifact TTL for type wandb-history")
+
+        if isinstance(ttl, ArtifactTTL):
+            self._ttl_duration_seconds = ttl.value
+        elif ttl.total_seconds() <= 0:
+            raise ValueError(
+                f"Artifact TTL Duration has to be positive. ttl: {ttl.total_seconds()}"
+            )
+        else:
+            self._ttl_duration_seconds = int(ttl.total_seconds())
```

Comment:
Nit: keep the same order of properties as in the constructor. This should go between `metadata` and `aliases`.